cmd/eval.go

```diff
@@ -86,5 +39,62 @@ func formatReplacements(replacementsInput map[string]string) map[string][]string
 // validateFile is a simple wrapper to ensure the input/output of valid YAML
 func validateFile(file []byte) error {
 	var holder interface{}
-	return yaml.Unmarshal(file, &holder)
+	err := yaml.Unmarshal(file, &holder)
+	if holder == nil {
+		return ErrEmptyFile
+	}
+	return err
 }
+
+var (
+	evalCmd = &cobra.Command{
+		Use:   "eval [file]",
+		Short: "Evaluate the combinations for a file at the given path",
+		Long: `
+Evaluate the combinations for a file at the given path. The file provided must be valid YAML.
+
+Note: the combo binary requires the --replacement flag to be explicitly set.
+
+The replacements flag allows users to specify a series of key value pairs in the form of KEY=VALUES.
+
+Example: combo eval -r REPLACE_ME=1,2,3 path/to/file
+	`,
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			replacements, err := cmd.Flags().GetStringToString("replacements")
+			if err != nil {
+				return fmt.Errorf("failed to access replacements flag: %w", err)
+			}
+
+			file, err := os.ReadFile(args[0])
+			if err != nil {
+				return fmt.Errorf("failed to read file specified: %w", err)
+			}
+
+			if err := validateFile(file); err != nil {
+				return fmt.Errorf("failed to validate file specified: %w", err)
+			}
```

Comment:
(out-of-scope) I'd like to see us refactor this driver function at some point and ensure that validateFile is actually stating the file and that the file exists as minimal checks while we iterate further. That could mean we also get access to existing stdlib methods for checking for empty files:

```go
f, err := os.Stat(file)
// handle error + os.IsNotExists(err)
if f.Size() == 0 {
  return ErrEmptyFile
```
I haven't dived into whether there's a simpler way to do this, but we should be able to clean up the process of validating the file, and opening that file for reading as this project gets further into development.